amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/hold/HoldServiceImpl.java

```diff
@@ -512,6 +513,40 @@ public void setHoldReason(NodeRef hold, String reason)
         }
     }
 
+    /**
+     * @see org.alfresco.module.org_alfresco_module_rm.hold.HoldService#setHoldDeletionReason(org.alfresco.service.cmr.repository.NodeRef, java.lang.String)
+     */
+    @Override
+    public void setHoldDeletionReason(NodeRef hold, String reason)
+    {
+        ParameterCheck.mandatory("hold", hold);
+        ParameterCheck.mandatory("reason", reason);
+
+        if (nodeService.exists(hold) && isHold(hold))
+        {
+            nodeService.setProperty(hold, PROP_HOLD_DELETION_REASON, reason);
+        }
+    }
+
+    /**
+     * @see org.alfresco.module.org_alfresco_module_rm.hold.HoldService#updateHold(org.alfresco.service.cmr.repository.NodeRef, java.lang.String, java.lang.String, java.lang.String) (org.alfresco.service.cmr.repository.NodeRef, java.lang.String, java.lang.String, java.lang.String)
+     */
+    @Override
+    public void updateHold(NodeRef hold, String name, String reason, String description)
+    {
+        ParameterCheck.mandatory("hold", hold);
+        ParameterCheck.mandatory("name", name);
+        ParameterCheck.mandatory("reason", reason);
+        ParameterCheck.mandatory("description", description);
```

Comment:
I think one of these is optional (maybe description?)

Reply:
Parameter check for the description removed.